### ISSUE TITLE
Remove possibility to change CMAKE_CXX_STANDARD

### ISSUE THREAD

**Issue by SunBlack** [PointCloudLibrary/pcl#3252]
#2690 added 

https://github.com/PointCloudLibrary/pcl/blob/a2bedc96e0b7d1a19a6e92ca9292afe009251ae5/CMakeLists.txt#L11

It is nice that we can switch C++14 and C++17, but we only support C++14, because we don't touch 3rd-Party code and with C++17 you get following error message:

```
/media/sf_pcl/surface/src/3rdparty/opennurbs/opennurbs_evaluate_nurbs.cpp:512:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register double alpha0;
  ^~~~~~~~~
/media/sf_pcl/surface/src/3rdparty/opennurbs/opennurbs_evaluate_nurbs.cpp:513:3: error: ISO C++17 does not allow 'register' storage class specifier [-Wregister]
  register double alpha1;
...
```

So we remove the option to allow the user to change the standard, as we know, only C++14 works and nothing other.

**Comment by taketwo** [comment]
Let's just do `s/register //g`.

**Comment by SunBlack** [comment]
Or can we update OpenNurbs? There `register` is already removed:

https://github.com/mcneel/opennurbs/blob/b844466e88fcf5a42c9d6d526abb3dcc44f939b5/opennurbs_evaluate_nurbs.cpp#L519-L523

**Comment by SergioRAgostinho** [comment]
Right now, it doesn't bother anyone. Since the our goal of allowing the user to select the standard >= C++14 has not changed, I would leave it untouched. Once we finish the correcting the changes that will allow PCL to be compiled with C++17, this will be one less thing to do.

**Comment by kunaltyagi** [comment]
What about updating the 3rd party libs?

**Comment by SunBlack** [comment]
I don't know about the license of 3rd party libs, if we are allowed to do so. In general I would prefer to integrate them via CMake and remove the code from our repo - but most of the 3rd party libs are not really CMake conform :-(

**Comment by SergioRAgostinho** [comment]
1. OpenNurbs: https://github.com/mcneel/opennurbs  the project is still active, the license is something akin to MIT. When [the code was added into our source tree](https://github.com/PointCloudLibrary/pcl/commit/3093623771895877c09cc4382e43e46032c8c022) it referred to the same organization. It would need some patching plus interest from the mother organization to modernize the build process.

2. Poisson Reconstruction: 
- Original project page: http://www.cs.jhu.edu/~misha/Code/PoissonRecon/Version12.00/
- CMake enabled repo tailored for MATLAB: https://github.com/daeyun/poisson-surface-reconstruction
- Original Commit: https://github.com/PointCloudLibrary/pcl/commit/3b20c6b6c116b859e42fc9ca59262a7099ecb9a4

Then again, why care about surface reconstruction as a point cloud focused library. Dense meshes are for newbies. :D (sarcasm)

**Comment by kunaltyagi** [comment]
There's also metslib in recognition in 3rd party. https://github.com/coin-or/metslib

Should I close issue #605, which is caused when someone uses the external metslib instead so we can track the 3rd party migration in one place?

**Comment by AlrIsmail** [comment]
I think this issue is also obsolete @larshg. Since PCL now requires C++17

**Comment by mvieth** [comment]
@AlrIsmail Yes, you are right. PCL requires C++17 (which is also the default), and is additionally compatible with (at least) C++20 and C++23. Thanks!
